### DISCUSSION TITLE
PDF

### DISCUSSION THREAD

**Discussion by Taghreed7878** [PaddlePaddle/PaddleOCR#12217] [Q&A]
Is there a way to make paddleocr object accepts bytes in case of PDF files like it accepts bytes in case of images?

**Comment by nullgogo** [comment]
用fitz这个库转img，用fitz.open(stream=pdf_bytes, filetype='bytes')这个方式读取，源码是直接open文件路径

希望能把PDF bytes files的方式也集成进去

**Comment by Taghreed7878** [comment]
But this will write the PDF bytes into a file, right?

**Comment by nullgogo** [comment]
> But this will write the PDF bytes into a file, right?

不需要写成file

参考以下代码，imgs可以直接作为入参

``` python
import fitz
from PIL import Image
imgs = []
with fitz.open(stream=pdf_bytes, filetype='bytes') as pdf:
    for pg in range(0, pdf.page_count):
        page = pdf[pg]
        mat = fitz.Matrix(2, 2)
        pm = page.get_pixmap(matrix=mat, alpha=False)

        # if width or height > 2000 pixels, don't enlarge the image
        if pm.width > 2000 or pm.height > 2000:
            pm = page.get_pixmap(matrix=fitz.Matrix(1, 1), alpha=False)

        img = Image.frombytes("RGB", [pm.width, pm.height], pm.samples)
        img = cv2.cvtColor(np.array(img), cv2.COLOR_RGB2BGR)
        imgs.append(img)
```

**Comment by nullgogo** [comment]
> > But this will write the PDF bytes into a file, right?
> 
> 不需要写成file
> 
> 参考以下代码，imgs可以直接作为入参
> 
> ```python
> import fitz
> from PIL import Image
> imgs = []
> with fitz.open(stream=pdf_bytes, filetype='bytes') as pdf:
>     for pg in range(0, pdf.page_count):
>         page = pdf[pg]
>         mat = fitz.Matrix(2, 2)
>         pm = page.get_pixmap(matrix=mat, alpha=False)
> 
>         # if width or height > 2000 pixels, don't enlarge the image
>         if pm.width > 2000 or pm.height > 2000:
>             pm = page.get_pixmap(matrix=fitz.Matrix(1, 1), alpha=False)
> 
>         img = Image.frombytes("RGB", [pm.width, pm.height], pm.samples)
>         img = cv2.cvtColor(np.array(img), cv2.COLOR_RGB2BGR)
>         imgs.append(img)
> ```

imgs作为一个list，det要传False... 这样似乎还不行，还得imgs[0],imgs[1]... 一页一页放进去，不懂是不是我搞错了

**Comment by Taghreed7878** [comment]
The issue with this approach is that each page will be processed separately I think, so the bboxes for each one will not be accumulated, right?

**Comment by BrownTen** [comment]
> > But this will write the PDF bytes into a file, right?
> 
> 不需要写成file
> 
> 参考以下代码，imgs可以直接作为入参
> 
> ```python
> import fitz
> from PIL import Image
> imgs = []
> with fitz.open(stream=pdf_bytes, filetype='bytes') as pdf:
>     for pg in range(0, pdf.page_count):
>         page = pdf[pg]
>         mat = fitz.Matrix(2, 2)
>         pm = page.get_pixmap(matrix=mat, alpha=False)
> 
>         # if width or height > 2000 pixels, don't enlarge the image
>         if pm.width > 2000 or pm.height > 2000:
>             pm = page.get_pixmap(matrix=fitz.Matrix(1, 1), alpha=False)
> 
>         img = Image.frombytes("RGB", [pm.width, pm.height], pm.samples)
>         img = cv2.cvtColor(np.array(img), cv2.COLOR_RGB2BGR)
>         imgs.append(img)
> ```

请问，为什么要做这个enlarge的操作？，虽然但是我直接get_pixmap()，的确效果不太好。
另外，他这个matrix的具体含义能讲一讲吗。是只是 resize还是做了相关增强？
